src/syscalls/stub.rs

```diff
@@ -105,7 +105,7 @@ pub fn load_cell_code(
 ) -> Result<usize, SysError> {
     let result = get().load_cell_code(buf_ptr, len, content_offset, content_size, index, source);
     match result {
-        Ok(()) => Ok(len),
+        Ok(_) => Ok(len),
```

Comment:
With this style of API, you should iterate based on the result of `load_cell_code`, and then fill in the actual data for each `MemoryPageResult` structure.